### ISSUE TITLE
DatePipe treats ISO-8601 YYYY-MM format as a date-time and adds timezone

### ISSUE THREAD

**Issue by pearlsze** [angular/angular#33944]
# 🐞 bug report

### Affected Package

 
@angular/common
### Is this a regression?

 No

### Description

 
I am trying to use the `DatePipe` with a string input that is in the ISO-8601 Year and Month format `YYYY-MM`. The DatePipe doc states that it supports ISO-8601 strings: https://github.com/angular/angular/blob/8.2.14/packages/common/src/pipes/date_pipe.ts#L160

The `DatePipe` incorrectly assumes the date is in GMT time. Then the resulting output is converted to the user's locale, which can cause off-by-1-day errors depending on your timezone.

E.g. `{{"2019-02" | date:'MMM. yyyy'}}`
Expected:  `Feb. 2019`
Actual: `Jan. 2019`

I don't see the same issue when using the other ISO-8601 format `YYYY-MM-DD`, there is a specific piece of code to handle timezone offsets for it: https://github.com/angular/angular/blob/8.2.14/packages/common/src/i18n/format_date.ts#L675. I think it should also apply to the `YYYY-MM` format.

## 🔬 Minimal Reproduction

https://stackblitz.com/edit/angular-jh7xqs 

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular: 8.2.9
... animations, bazel, common, compiler, compiler-cli, core
... forms, platform-browser, platform-browser-dynamic, router
</code></pre>

**Comment by petebacondarwin** [comment]
This looks like a valid bug. Here is where it was fixed for full dates: https://github.com/angular/angular/commit/2aba8b0ff2c5d6979802b96842aceccad3d5716d

**Comment by petebacondarwin** [comment]
@pearlsze - I can no longer reproduce this. I can't see anything that has changed in the codebase to fix this. But then I wonder if there is something about the local timezone that makes the difference. Can you reproduce it with the latest Angular?

**Comment by pearlsze** [comment]
@petebacondarwin Hi, I made a new [stackblitz](https://stackblitz.com/edit/angular-ivy-v1ddqc?file=src/app/app.component.html) with the latest angular 11.1.1, and still see the issue.

The output I see (with the first one being incorrect). 
```
With YYYY-MM format
Input: 2019-02
Output: Jan. 2019

Specifying pipe to use 'GMT' timezone
Input: 2019-02
Output: Feb. 2019

With YYYY-MM-DD format
Input: 2019-02-01
Output: Feb. 2019

```

**Comment by petebacondarwin** [comment]
@pearlsze what timezone are you in? I am in London UK (GMT) and I don't see `Jan. 2019` I see `Feb. 2019`.
I tried overriding the timezone in Chrome but whatever time-zone I choose I get the same result.

**Comment by pearlsze** [comment]
@petebacondarwin I am in PST! Using the location override in Chrome (assuming you are too), "San Francisco" reproduces the issue, but for "London" I no longer see the issue.

**Comment by petebacondarwin** [comment]
Hmm, I tried "Mountain View"... let me try again.

**Comment by petebacondarwin** [comment]
OK!

In my local timezone `new Date('2019-02')` gives me `Fri Feb 01 2019 00:00:00 GMT+0000 (Greenwich Mean Time)`
And in San Francisco timezone `new Date('2019-02')` gives me `Thu Jan 31 2019 16:00:00 GMT-0800 (Pacific Standard Time)`
